### ISSUE TITLE
Create first pass of SimpleNNs.jl

### ISSUE THREAD

**Issue by JamieMair** [JamieMair/SimpleNNs.jl#1]
Create a package that allows you to create a neural network:
- [x] Specify NN structure like Flux.chain
- [x] Give a flat parameter vector for the neural network
- [x] Preallocate memory for a given batch size
- [x] Work on generic arrays, GPU or CPU
- [ ] ~~Custom loss functions for backpropagation~~
